src/server/ua_securechannel_manager.c

```diff
@@ -210,10 +211,9 @@ UA_SecureChannelManager_renew(UA_SecureChannelManager* cm, UA_SecureChannel *cha
     UA_ByteString_deleteMembers(&channel->remoteNonce);
     UA_ByteString_copy(&request->clientNonce, &channel->remoteNonce);
 
-    const size_t keyLength = channel->securityPolicy->symmetricModule.cryptoModule.
-        getLocalEncryptionKeyLength(channel->securityPolicy, channel->channelContext);
     UA_ByteString_deleteMembers(&channel->localNonce);
-    UA_SecureChannel_generateNonce(channel, keyLength, &channel->localNonce);
```

Comment:
This can be simplified to `UA_SecureChannel_generateNonce(channel, &channel->localNonce)`.
The SecureChannel has access to the policy internally.

Then we don't have to include `ua_plugin_securitypolicy.h` here. And the code remains decoupled.